### PR TITLE
Patch components

### PR DESCRIPTION
**Pull request by vishwajeetkumar93**
# Pull Request      

## What does this PR do?

Moved `components` folder directly under primeqa. Makes refactoring changes to reflect the new folder structure.

### CODE REVIEW

File: primeqa/components/retriever/sparse.py
Comment (bhavani105) [inline primeqa/components/retriever/sparse.py:5]:
```suggestion
from primeqa.components.base import Retriever
```

File: primeqa/services/grpc_server/protos/indexer.proto
Comment (bhavani105) [inline primeqa/services/grpc_server/protos/indexer.proto:16]: Changes to the proto needs to be in sync with the proto files in the primeqa-orchestrator

**Review by bhavani105 — COMMENTED** [review]
LGTM left some comments the main one beind
changes to services protos must be synch with primeqa-orchestrator
https://github.com/primeqa/primeqa-orchestrator/tree/main/orchestrator/integrations/primeqa/protos

**Review by jdpsen — COMMENTED** [review]
Should we add few comments to describe the base classes are to be extended by each new component ?

LGTM, in general.

**Comment by kpfadnis** [comment]
@jdpsen @vishwajeetkumar93 I have updated and tested `primeqa` repo's service layer from `primeqa-orchestrator`. All the necessary changes are now part of the PR. 

@seirasto Please check `mrc.py` to validate temporary fix I put in for `transformers == 4.24.0` 

@avisil @bhavani105 Please do a detailed review to ensure I haven't missed anything.

File: primeqa/mrc/trainers/mrc.py
Comment (seirasto) [inline primeqa/mrc/trainers/mrc.py:41]: This solution looks nice and clean but it needs to be modified for the run_mrc.py case where the args are already there.

Comment (jdpsen) [reply primeqa/mrc/trainers/mrc.py]: @seirasto @kpfadnis : who is fixing this ? This is a pre-req for Tapex and hence I need both your sign-offs from services and primeqa mrc side, before I can approve/merge it.

Comment (kpfadnis) [reply primeqa/mrc/trainers/mrc.py]: @seirasto Made changes as per our conversation. Please take a look when you have a minute.

@jdpsen Once @seirasto reviews, you might can do full review for the PR.

Comment (seirasto) [reply primeqa/mrc/trainers/mrc.py]: LGTM! Thanks @kpfadnis

**Review by jdpsen — APPROVED** [review]
Code changes look good to me.

File: primeqa/components/base.py
Comment (seirasto) [inline primeqa/components/base.py:145]:
```suggestion
        Return this retriever engine type. Must match with the retriever that will be used to query the index.
```

File: notebooks/ir/dense/dense_ir_pipeline.ipynb
Comment (avisil) [inline notebooks/ir/dense/dense_ir_pipeline.ipynb:589]: I'm not a vscode user - is this hash necessary to be kept?

**Review by avisil — COMMENTED** [review]
LGTM. Left a couple of minor comments. I'll approve once:
 - others' comments are addressed. Please confirm with me.
 - branch is rebased with main and all is okay. It seems stale at the moment.

File: notebooks/ir/dense/dense_ir_pipeline.ipynb
Comment (vishwajeetkumar93) [reply notebooks/ir/dense/dense_ir_pipeline.ipynb:589]: I think this gets automatically generated when you run the notebook in vscode. It will keep on changing its a sort of meta data attached to ipynb.

**Comment by kpfadnis** [comment]
@jdpsen @vishwajeetkumar93 if either of you guys are merging it, make sure you squash and merge since there are too many minor fix commits in this PR.

File: primeqa/services/grpc_server/protos/indexer.proto
Comment (jdpsen) [reply primeqa/services/grpc_server/protos/indexer.proto:33]: to be raised as a separate PR in primeqa-orchestrator cc: @kpfadnis

File: primeqa/services/grpc_server/protos/indexer.proto
Comment (jdpsen) [reply primeqa/services/grpc_server/protos/indexer.proto:16]: Fixed in commit 531f701cee0159152fa6927da2d60170f682949e

File: primeqa/services/grpc_server/protos/retriever.proto
Comment (jdpsen) [reply primeqa/services/grpc_server/protos/retriever.proto:15]: fixed in 531f701cee0159152fa6927da2d60170f682949e

File: primeqa/services/grpc_server/protos/reader.proto
Comment (jdpsen) [reply primeqa/services/grpc_server/protos/reader.proto:14]: fixed in commit 531f701cee0159152fa6927da2d60170f682949e

**Comment by jdpsen** [comment]
> @jdpsen @vishwajeetkumar93 I have updated and tested `primeqa` repo's service layer from `primeqa-orchestrator`. All the necessary changes are now part of the PR.
> Thanks.
> @seirasto Please check `mrc.py` to validate temporary fix I put in for `transformers == 4.24.0`
> Fixed in commit 6619f2411b1028e0c99446cc46df7d238734def5
> @avisil @bhavani105 Please do a detailed review to ensure I haven't missed anything.

**Review by avisil — APPROVED** [review]
LGTM - approved.

**Pull request merged**
